source/lib.ts

```diff
@@ -136,12 +171,13 @@ const parseOptions = (passedOptions: Partial<Options>): Configuration => {
 			response.statusCode < 400,
 		skip: (_request: Request, _response: Response): boolean => false,
 		keyGenerator(request: Request, _response: Response): string {
-			if (!request.ip) {
-				console.error(
-					'WARN | `express-rate-limit` | `request.ip` is undefined. You can avoid this by providing a custom `keyGenerator` function, but it may be indicative of a larger issue.',
-				)
-			}
+			// Run the validation checks on the IP and headers to make sure everything
+			// is working as intended.
+			validateIp(config.validation, request.ip)
+			validateTrustProxy(config.validation, request)
+			validateXForwardedForHeader(config.validation, request)
```

Comment:
Looks like we also need to add `checkIfIpIsUndefined(request.ip)` here. Alternatively, it could just be run as part of `validateIp()` and it could return early - if it fails checkIfIpIsUndefined, then it's guarenteed to fail validateIp.